Cognite.Config/KeyVault.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Runtime.Serialization;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+using Cognite.Extractor.Common;
+using Cognite.Extractor.Configuration;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+
+namespace Cognite.Extractor.KeyVault
+{
+    /// <summary>
+    /// Method used to authenticate against azure.
+    /// </summary>
+    public enum KeyVaultAuthenticationMethod
+    {
+        /// <summary>
+        /// Authenticate using client secret credentials.
+        /// </summary>
+        [EnumMember(Value = "client-secret")]
+        ClientSecret,
+        /// <summary>
+        /// Authenticate using machine default credentials.
+        /// </summary>
+        [EnumMember(Value = "default")]
+        Default,
+    }
+
+
+    /// <summary>
+    /// Configuration for azure key vault.
+    /// </summary>
+    public class KeyVaultConfig
+    {
+        /// <summary>
+        /// How to authenticate against key vault.
+        /// </summary>
+        public KeyVaultAuthenticationMethod? AuthenticationMethod { get; set; }
+        /// <summary>
+        /// Azure key vault name.
+        /// </summary>
+        public string? VaultName { get; set; }
```

Comment:
Nitpick: This property is called [`keyvault-name` in the python utils](https://github.com/cognitedata/python-extractor-utils/blob/master/cognite/extractorutils/configtools/loaders.py#L73), we should use the same name here